hw3/03-react/game-of-thrones/src/index.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import ReactDOM from "react-dom/client";
+import "./index.css";
+import App from "./App";
+import "./App.css";
+import reportWebVitals from "./reportWebVitals";
+import { BrowserRouter } from "react-router-dom";
+import "bootstrap/dist/css/bootstrap.min.css";
+import Home from "./views/Home";
+import Search from "./views/Search";
+import Houses from "./views/Houses";
+import { Route, Routes } from "react-router-dom";
+
+const root = ReactDOM.createRoot(document.getElementById("root"));
+root.render(
+  <BrowserRouter>
+    <App />
+    <Routes>
+      <Route path="/" element={<Home />} />
+      <Route path="/search" element={<Search />} />
+      <Route path="/houses" element={<Houses />} />
+    </Routes>
```

Comment:
Generally, it is standard for this to be in App.js, not index.js. The index.js should be very simple.

Also, Fetching the data (calling the exact same API) twice, in both the Houses and Search components is inefficient. The data they both rely on is the same, and additionally both of these components have the same parent (should be App.js).

So why not let the data flow from parent to child?

It would make a lot more sense to call the API once in the App component, then pass the data as props to the houses and search components. This would be more efficient since the API would only need to be called once, but the data would be available across the components that need it.

In summary, you can just fetch the data once and then pass it to child components via props.